File: apps/matriz-desktop/src-tauri/src/catalog.rs
#[derive(Clone, Copy, Debug)]
pub struct AppDefinition {
    pub id: &'static str,
    pub label: &'static str,
    pub package_name: &'static str,
    pub directory: &'static str,
    pub port: u16,
}

#[derive(Clone, Copy, Debug)]
pub struct GateDefinition {
    pub id: &'static str,
    pub script: &'static str,
}

#[derive(Clone, Copy, Debug)]
pub enum QuickTarget {
    Workspace,
    Terminal,
    Url(&'static str),
}

#[derive(Clone, Copy, Debug, PartialEq, Eq)]
pub enum ManagedOperationKind {
    Command,
    NativeInstall,
    NativeStart,
}

#[derive(Clone, Debug, PartialEq, Eq)]
pub struct ManagedOperationDefinition {
    pub id: String,
    pub title: String,
    pub kind: ManagedOperationKind,
    pub program: Option<String>,
    pub args: Vec<String>,
}

const APPS: [AppDefinition; 9] = [
    AppDefinition {
        id: "matriz-hub",
        label: "Matriz Hub",
        package_name: "@matriz/app-matriz-hub",
        directory: "apps/matriz-hub",
        port: 3000,
    },
    AppDefinition {
        id: "spot",
        label: "Spot",
        package_name: "@matriz/app-spot",
        directory: "apps/spot",
        port: 3001,
    },
    AppDefinition {
        id: "matriz-admin",
        label: "Matriz Admin",
        package_name: "@matriz/app-matriz-admin",
        directory: "apps/matriz-admin",
        port: 3002,
    },
    AppDefinition {
        id: "contracts",
        label: "Contracts",
        package_name: "@matriz/app-contracts",
        directory: "apps/contracts",
        port: 3003,
    },
    AppDefinition {
        id: "willdash",
        label: "Willdash",
        package_name: "@matriz/app-willdash",
        directory: "apps/willdash",
        port: 3004,
    },
    AppDefinition {
        id: "matriz-workbench",
        label: "Workbench",
        package_name: "@matriz/app-matriz-workbench",
        directory: "apps/matriz-workbench",
        port: 3005,
    },
    AppDefinition {
        id: "sites",
        label: "Sites",
        package_name: "@matriz/app-sites",
        directory: "apps/sites",
        port: 3006,
    },
    AppDefinition {
        id: "matrizlib",
        label: "MatrizLib",
        package_name: "@matriz/app-matrizlib",
        directory: "apps/matrizlib",
        port: 3007,
    },
    AppDefinition {
        id: "seumei",
        label: "Seumei",
        package_name: "@matriz/app-seumei",
        directory: "apps/seumeiapp",
        port: 3008,
    },
];

const GATES: [GateDefinition; 4] = [
    GateDefinition {
        id: "typecheck",
        script: "typecheck",
    },
    GateDefinition {
        id: "lint",
        script: "lint",
    },
    GateDefinition {
        id: "test:smoke",
        script: "test:smoke",
    },
    GateDefinition {
        id: "prisma:validate",
        script: "prisma:validate",
    },
];

pub fn app_definition(id: &str) -> Result<AppDefinition, String> {
    APPS.iter()
        .copied()
        .find(|definition| definition.id == id)
        .ok_or_else(|| format!("Unknown Matriz app: {id}"))
}

pub fn gate_definition(id: &str) -> Result<GateDefinition, String> {
    GATES
        .iter()
        .copied()
        .find(|definition| definition.id == id)
        .ok_or_else(|| format!("Unknown validation gate: {id}"))
}

pub fn quick_target(id: &str) -> Result<QuickTarget, String> {
    match id {
        "workspace" => Ok(QuickTarget::Workspace),
        "terminal" => Ok(QuickTarget::Terminal),
        "hub" => Ok(QuickTarget::Url("http://localhost:3000")),
        "matrizlib" => Ok(QuickTarget::Url("http://localhost:3007")),
        "workbench" => Ok(QuickTarget::Url("http://localhost:3005")),
        _ => Err(format!("Unknown quick target: {id}")),
    }
}

pub fn apps() -> &'static [AppDefinition] {
    &APPS
}

pub fn managed_operation(id: &str) -> Result<ManagedOperationDefinition, String> {
    if let Some(app_id) = id
        .strip_prefix("app.")
        .and_then(|value| value.strip_suffix(".web"))
    {
        let app = app_definition(app_id)?;
        return Ok(ManagedOperationDefinition {
            id: id.to_owned(),
            title: format!("{} / WEB", app.id.to_uppercase()),
            kind: ManagedOperationKind::Command,
            program: Some("pnpm.cmd".into()),
            args: ["--filter", app.package_name, "dev"]
                .map(str::to_owned)
                .into(),
        });
    }
    if let Some(gate_id) = id.strip_prefix("gate.") {
        let gate = gate_definition(gate_id)?;
        return Ok(ManagedOperationDefinition {
            id: id.to_owned(),
            title: format!("{} / GATE", gate.id.to_uppercase()),
            kind: ManagedOperationKind::Command,
            program: Some("pnpm.cmd".into()),
            args: ["run", gate.script].map(str::to_owned).into(),
        });
    }
    match id {
        "app.matriz-admin.native.build" => Ok(ManagedOperationDefinition {
            id: id.to_owned(),
            title: "MATRIZ ADMIN / BUILD".into(),
            kind: ManagedOperationKind::Command,
            program: Some("powershell.exe".into()),
            args: [
                "-NoProfile",
                "-NonInteractive",
                "-ExecutionPolicy",
                "Bypass",
                "-File",
                "apps/matriz-desktop/scripts/package-matriz-admin.ps1",
            ]
            .map(str::to_owned)
            .into(),
        }),
        "app.matriz-admin.native.install" => Ok(ManagedOperationDefinition {
            id: id.to_owned(),
            title: "MATRIZ ADMIN / INSTALL".into(),
            kind: ManagedOperationKind::NativeInstall,
            program: None,
            args: Vec::new(),
        }),
        "app.matriz-admin.native.start" => Ok(ManagedOperationDefinition {
            id: id.to_owned(),
            title: "MATRIZ ADMIN / NATIVE".into(),
            kind: ManagedOperationKind::NativeStart,
            program: None,
            args: Vec::new(),
        }),
        _ => Err(format!("Unknown managed operation: {id}")),
    }
}

File: apps/matriz-desktop/src/application/catalog.ts
import type { DesktopAppId, GateId, ManagedOperationId, QuickTargetId } from "../domain/types"

export interface DesktopAppDefinition {
  readonly id: DesktopAppId
  readonly label: string
  readonly packageName: string
  readonly port: number
}

export const MATRIZ_DESKTOP_APPS: readonly DesktopAppDefinition[] = Object.freeze([
  { id: "matriz-hub", label: "Hub", packageName: "@matriz/app-matriz-hub", port: 3000 },
  { id: "spot", label: "Spot", packageName: "@matriz/app-spot", port: 3001 },
  { id: "matriz-admin", label: "Matriz Admin", packageName: "@matriz/app-matriz-admin", port: 3002 },
  { id: "contracts", label: "Contracts", packageName: "@matriz/app-contracts", port: 3003 },
  { id: "willdash", label: "Willdash", packageName: "@matriz/app-willdash", port: 3004 },
  {
    id: "matriz-workbench",
    label: "Workbench",
    packageName: "@matriz/app-matriz-workbench",
    port: 3005,
  },
  { id: "sites", label: "Sites", packageName: "@matriz/app-sites", port: 3006 },
  { id: "matrizlib", label: "MatrizLib", packageName: "@matriz/app-matrizlib", port: 3007 },
  { id: "seumei", label: "Seumei", packageName: "@matriz/app-seumei", port: 3008 },
  { id: "health", label: "Health", packageName: "@matriz/app-health", port: 3010 },
  { id: "matriz-ops", label: "Matriz Ops", packageName: "@matriz/app-matriz-ops", port: 3011 },
  { id: "matriz-pay", label: "Matriz Pay", packageName: "@matriz/app-matriz-pay", port: 3012 },
  { id: "matriz-client-admin", label: "Client Admin", packageName: "@matriz/app-client-admin", port: 3013 },
])

export const GATES: readonly { readonly id: GateId; readonly label: string }[] = Object.freeze([
  { id: "typecheck", label: "Types" },
  { id: "lint", label: "Lint" },
  { id: "test:smoke", label: "Smoke" },
  { id: "prisma:validate", label: "Prisma" },
])

export const QUICK_TARGETS: readonly { readonly id: QuickTargetId; readonly label: string }[] =
  Object.freeze([
    { id: "workspace", label: "Files" },
    { id: "terminal", label: "Terminal" },
    { id: "hub", label: "Hub" },
    { id: "matrizlib", label: "MatrizLib" },
    { id: "workbench", label: "Workbench" },
  ])

export const MANAGED_OPERATIONS: readonly {
  readonly id: ManagedOperationId
  readonly label: string
}[] = Object.freeze([
  ...MATRIZ_DESKTOP_APPS.map((app) => ({
    id: `app.${app.id}.web` as const,
    label: `${app.label} / Web`,
  })),
  { id: "app.matriz-admin.native.build", label: "Matriz Admin / Gerar" },
  { id: "app.matriz-admin.native.install", label: "Matriz Admin / Instalar" },
  { id: "app.matriz-admin.native.start", label: "Matriz Admin / Nativo" },
  ...GATES.map((gate) => ({ id: `gate.${gate.id}` as const, label: gate.label })),
])
